include/libtorrent/session.hpp

```diff
@@ -147,23 +175,37 @@ namespace libtorrent
 	{
 	public:
 
+		session(session_params params = session_params())
+			: session_handle(nullptr)
+		{
+			TORRENT_CFG();
+			start(std::move(params), nullptr);
+		}
+
+		session(session_params params, io_service& ios)
+			: session_handle(nullptr)
+		{
+			TORRENT_CFG();
+			start(std::move(params), &ios);
+		}
+
 		// Constructs the session objects which acts as the container of torrents.
 		// It provides configuration options across torrents (such as rate limits,
 		// disk cache, ip filter etc.). In order to avoid a race condition between
 		// starting the session and configuring it, you can pass in a
 		// settings_pack object. Its settings will take effect before the session
 		// starts up.
-		// 
+		//
 		// The ``flags`` parameter can be used to start default features (upnp &
 		// nat-pmp) and default plugins (ut_metadata, ut_pex and smart_ban). The
 		// default is to start those features. If you do not want them to start,
 		// pass 0 as the flags parameter.
-		session(settings_pack pack = settings_pack()
-			, int flags = start_default_features | add_default_plugins)
+		session(settings_pack pack
+			, int flags = session_params::default_flags)
```

Comment:
thinking about it, keeping this flag in the current form could create dangerous inconsistencies at creation time, since the `params.extenstions` could have the default plugins already, re-thinking this...